### PR TITLE
Refactor to use AyahShape

### PR DESCRIPTION
**Pull request by piraka9011**

### CODE REVIEW

File: src/transcribe.js [hunk 1]
```diff
@@ -82,11 +82,10 @@ export class Transcriber {
                     ];
 
                     this.getAyat(ayatList, (err, results) => {
-                        let ayahText = results[0].ayahText;
-                        this.onAyahFound(surahNum, ayahNum, ayahText);
+                        this.onAyahFound(results[0]);
 
-                        this.currentAyah = ayahText;
-                        this.nextAyah = results === null ? null : results[1].ayahText;
+                        this.currentAyah = results[0];
```
Comment (fdalvi) [inline src/transcribe.js]: Now that we are storing the entire object as `this.currentAyah`, I think we can safely remove `this.currentSurahNum` and `this.currentAyahNum` and just use the `chapter_id`/`verse_number` from the `this.currentAyah` object

**Comment by fdalvi** [comment]
BTW I can also work on this tonight if you don't have many free cycles, should not take too long because I think I know the issue and the fix already for the tests

**Comment by piraka9011** [comment]
> BTW I can also work on this tonight if you don't have many free cycles, should not take too long because I think I know the issue and the fix already for the tests

Yes please :)

**Comment by fdalvi** [comment]
Changes done, tests pass now and also made the code a bit better. Take a look @piraka9011 @Ananasship 

There is one thing that I still dont like, our `ayahFound` event returns a message that looks like:
```
{
  ayahShape: {
    chapter_id: ...,
    verse_number: ...,
    ....
  }
}
```
but `matchFound` returns:
```
{
  match: {
    chapter_id: ...,
    verse_number: ...,
    ....
  },
  numWords: ...
}
```

In one case the ayah object is referenced by the key `ayahShape`, while in the other its referenced as `match`. I didn't change this because it'll break the frontend. Lmk what you guys think.

**Comment by Ananasship** [comment]
+1 for consistency in naming them. Can we create an issue for this Fahim?

File: src/transcribe.js [hunk 2]
```diff
@@ -7,11 +7,11 @@ import * as transcription_constants from '../config/transcriptionConstants';
 
 export class Transcriber {
     constructor(onAyahFound, onMatchFound) {
-        // Current Ayah trackers
-        this.currentSurahNum = null;
-        this.currentAyahNum = null;
-
-        // Textual representations of ayah
+        // Object representations of ayat containing
+        //  - chapter_id: Juz number
```
Comment (Ananasship) [inline src/transcribe.js]: Is this Juz number or Surah number?

Comment (piraka9011) [reply src/transcribe.js]: chapter_id should be surah

**Comment by piraka9011** [comment]
> +1 for consistency in naming them. Can we create an issue for this Fahim?

I can take a look at this when I'm adding the changes for our API.
@fdalvi you can assign that issue to me.

File: src/transcribe.js (hunk 2 above)
Comment (fdalvi) [reply src/transcribe.js]: ahh good catch!

**Pull request merged**